configuration/secrets.rst

```diff
@@ -0,0 +1,312 @@
+.. index::
+    single: Secrets
+
+How to Keep Sensitive Information Secret
+========================================
+
+In :doc:`/configuration` and :doc:`/configuration/environment_variables`, you
+learned how to manage your application configuration. In this article you'll
+learn how to easily and saftely configure your application with sensitive
+information such as credentials, passwords, tokens, API keys without exposing
+them.
+
+.. _secrets-configuration:
+
+Configuration
+-------------
+
+In order to use secrets you have to enable the feature in the framework's
+configuration:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            secrets:
+                # encrypted_secrets_dir: '%kernel.project_dir%/config/secrets/%kernel.environment%/'
+                # encryption_key: '%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key'
+
+    .. code-block:: xml
+
+            <!-- config/packages/framework.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:framework="http://symfony.com/schema/dic/framework"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/framework https://symfony.com/schema/dic/framework/framework-1.0.xsd"
+            >
+                <framework:config secret="%env(APP_SECRET)%">
+                    <framework:secrets/>
+                </framework:config>
+            </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        $container->loadFromExtension('framework', [
+            'secrets' => true,
+        ]);
+
+.. _secrets-generate-key:
+
+Generate an Encryption Key
+--------------------------
+
+Before creating a new **secret**, you need to create an **encryption key**.
+This can be done with the provided command ``secrets:generate-key``.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:generate-key
+
+This command generates a new **encryption key** in
+``%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key``
+
+.. note::
+
+    In order to use Symfony's built-in secret storage, you will need the
+    `libsodium`_ PHP extension or use the `sodium_compat`_ package.
+
+Symfony generates a key with a symetrics algorithm, meaning that this key could
+be used to both encrypt **and** decrypt secrets. The number of people who
+possess this key should be as small as possible.
+
+.. caution::
+
+    This file is sensitive and **must not** be commited nor publicly shared.
+    Every developpers and CI don't need that key. If the **encryption key**
```

Comment:
`Every developpers and CI don't need that key.` -> `Your team developers and Continuous Integration services don't need that key.`